developers toolbox (#153)
iExecBlockchainComputing/documentation: for-developers/toolbox/iexec-subgraph.md

## GraphQL API

The subgraph can be queried through the [Graph*i*QL interface](https://thegraph.bellecour.iex.ec/subgraphs/name/bellecour/poco-v5/graphql)

The GraphQL API is publicly available:

- for queries: `https://thegraph.bellecour.iex.ec/subgraphs/name/bellecour/poco-v5`
- for subscriptions: `wss://ws-thegraph.bellecour.iex.ec/subgraphs/name/bellecour/poco-v5`

{% hint style="warning" %}

This subgraph is hosted by iExec and shared across all developers.

Keep in mind that you must use efficient calls and caching policy when querying this subgraph to avoid being rate limited.

{% endhint %}

## Examples queries

### Protocol metrics

```graphql
{
  protocol(id: "iExec") {
    appsCount
    datasetsCount
    workerpoolsCount
    dealsCount
    tasksCount
    completedTasksCount
    claimedTasksCount
  }
}
```

[Try this query](https://thegraph.bellecour.iex.ec/subgraphs/name/bellecour/poco-v5/graphql?query=%7B%0A++protocol%28id%3A+%22iExec%22%29+%7B%0A++++appsCount%0A++++datasetsCount%0A++++workerpoolsCount%0A++++dealsCount%0A++++tasksCount%0A++++completedTasksCount%0A++++claimedTasksCount%0A++%7D%0A%7D)

> ### Application latests usages

Reviewer: ```suggestion
### Application latest usages
```

## FAQ

Here are some frequently asked questions about iExec's subgraph.

You may also find useful information on these resources:

- [GraphQL documentation](https://graphql.org/learn/)
- [The Graph documentation](https://thegraph.com/docs/)

### How to find an item by it's ethereum address?

Although an ethereum address has multiple valid formats, The Graph requires lowercase addresses.

- :heavy_check_mark: lowercase `0xf048ef3d7e3b33a465e0599e641bb29421f7df92`
- :x: checksum `0xF048eF3d7E3B33A465E0599E641BB29421f7Df92`
- :x: uppercase `0XF048EF3D7E3B33A465E0599E641BB29421F7DF92`

### How to get more than 100 items?

By default, query responses are limited to 100 items per collection. If you want to receive more, you can go up to 1000 items per collection and beyond that, you can paginate with:

```graphql
someCollection(first: 1000, skip: <number>) { ... }
```

### Can I rely on a subgraph to check the state of the protocol?

Subgraphs are built to quickly access a representation of onchain state through a friendly API.

> Under the hood, a subgraph is a centralized system watching the blockchain events and indexing data in a database. Like any system, it may be subjects to issues (bugs, sync problems...) and reflects an erroneous or outdated state of the onchain data.

Reviewer: like `and it may reflect`
```suggestion
Under the hood, a subgraph is a centralized system watching the blockchain events and indexing data in a database. Like any system, it may be subjects to issues (bugs, sync problems...) and reflect an erroneous or outdated state of the onchain data.
```

Author: `subjects` as well

```suggestion
Under the hood, a subgraph is a centralized system watching the blockchain events and indexing data in a database. Like any system, it may be subject to issues (bugs, sync problems...) and reflect an erroneous or outdated state of the onchain data.
```
